### ISSUE TITLE
Question: config file to override default 'MD029' rule

### ISSUE THREAD

**Issue by jeff1evesque** [markdownlint/markdownlint#113]
I have the following [`.travis.yml`](https://github.com/jeff1evesque/machine-learning/blob/2b2fc88f56800beb8baed0c076e65737544835b9/.travis.yml#L59), which executes the `mdl . style 'test/config/markdown.md'`:

```
sudo: required

language: python

## allows 'docker' command
services:
  - docker

## build docker container for unit testing.
before_install:
  - docker ps -a
  - docker build -t ubuntu/trusty .

## install packages for linting
#
#  @trusty-backports, allows additional packages (i.e. shellcheck) to be installed
#
#  Note: unit testing is performed within the docker container, which is
#        defined from the Dockerfile.
install:
  - sudo add-apt-repository "deb http://archive.ubuntu.com/ubuntu/ trusty-backports restricted main universe"
  - pip install flake8
  - npm install -g jshint
  - npm install jscs -g
  - gem install scss_lint
  - npm install -g bootlint
  - gem install puppet-lint
  - gem install r10k
  - npm install jsonlint -g
  - gem install csvlint
  - sudo apt-get update
  - sudo apt-get install libxml2-utils
  - gem install mdl
  - sudo apt-get install shellcheck

## implement linting
#
#  @puppet-lint, the following issue needs to be checked if it has been
#      resolved, so the '--no-ensure_first_param-check' abusive flag
#      implementation (below) can be removed:
#
#      - https://github.com/rodjek/puppet-lint/issues/410
#
#  Note: unit testing is performed within the docker container, which is
#        defined from the Dockerfile.
script:
  - flake8 .
  - jshint --verbose src/js/*.js --config test/config/jshint.json --exclude-path test/config/.jshintignore
  - jscs . --config test/config/.jscs.json
  - scss-lint src/scss/*.scss
  - bootlint interface/templates/*.html
  - puppet-lint --no-ensure_first_param-check puppet/manifests
  - (cd puppet && r10k puppetfile check)
  - (cd puppet/scripts/ && r10k puppetfile check)
  - find interface/static/data/json -name '*.json' -exec jsonlint -q {} \; -print
  - find interface/static/data/csv -name '*.csv' -exec csvlint {} \;
  - xmllint --noout interface/static/data/xml/*.xml --schema interface/static/data/xml/dataset.xsd
  - xmllint --noout interface/static/data/xml/*.xsd
  - mdl . style 'test/config/markdown.md'
  - find . -type f -exec grep -l '^#!/bin/bash' {} + | xargs shellcheck
  - find . -name '*.sh' | xargs shellcheck
```

The corresponding [`markdown.md`](https://github.com/jeff1evesque/machine-learning/blob/8c9e915da6600735287b583b58c85a334e82d16f/test/config/markdown.rb):

``` markdown
rule 'MD029', :style => :ordered
```

However, my travis implementation gives me the following [error](https://travis-ci.org/jeff1evesque/machine-learning/builds/89944758#L1050-L1061):

``` bash
...
/usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/doc.rb:44:in `read': No such file or directory @ rb_sysopen - style (Errno::ENOENT)
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/doc.rb:44:in `new_from_file'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl.rb:61:in `block in run'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl.rb:59:in `each'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl.rb:59:in `run'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/bin/mdl:10:in `<top (required)>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `load'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `<main>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `eval'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `<main>'
The command "mdl . style 'test/config/markdown.md'" exited with 1.
...
```

Do I have to adjust my [`markdown.md`](https://github.com/jeff1evesque/machine-learning/blob/8c9e915da6600735287b583b58c85a334e82d16f/test/config/markdown.rb), or have to define a corresponding `.mdlrc`?  If so, how do I reference it within my [`markdown.md`](https://github.com/jeff1evesque/machine-learning/blob/8c9e915da6600735287b583b58c85a334e82d16f/test/config/markdown.rb), or within my [`.travis.yml`](https://github.com/jeff1evesque/machine-learning/blob/2b2fc88f56800beb8baed0c076e65737544835b9/.travis.yml#L59)?

**Comment by jeff1evesque** [comment]
When I change the following line from [`.travis.yml`](https://github.com/jeff1evesque/machine-learning/blob/2b2fc88f56800beb8baed0c076e65737544835b9/.travis.yml#L59):

```
...
  - mdl . style 'test/config/markdown.md'
...
```

to the following:

``` markdown
...
  - mdl . --style 'test/config/markdown.md'
...
```

my travis implementation gives me the following [error](https://travis-ci.org/jeff1evesque/machine-learning/builds/90013269#L948-L958):

``` bash
...
$ mdl . --style 'test/config/markdown.md'

/usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/style.rb:45:in `read': No such file or directory @ rb_sysopen - test/config/markdown.md (Errno::ENOENT)
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/style.rb:45:in `load'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl.rb:16:in `run'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/bin/mdl:10:in `<top (required)>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `load'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `<main>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `eval'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `<main>'
The command "mdl . --style 'test/config/markdown.md'" exited with 1.
...
```

**Comment by jeff1evesque** [comment]
When I [change](https://github.com/jeff1evesque/machine-learning/commit/6635efd92424d00695b8ef7cc764a5c3d0aecf40) the following line from `.travis.yml`:

```
...
  - mdl . --style 'test/config/markdown.md'
...
```

to the following:

``` markdown
...
  - mdl . --style test/config/markdown.md
...
```

my travis implementation gives me the following [error](https://travis-ci.org/jeff1evesque/machine-learning/builds/90017927#L948-L958):

``` bash
...
$ mdl . --style test/config/markdown.md

/usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/style.rb:45:in `read': No such file or directory @ rb_sysopen - test/config/markdown.md (Errno::ENOENT)
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/style.rb:45:in `load'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl.rb:16:in `run'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/bin/mdl:10:in `<top (required)>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `load'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `<main>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `eval'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `<main>'
The command "mdl . --style test/config/markdown.md" exited with 1.
...
```

**Comment by jeff1evesque** [comment]
I changed the [`markdown.md`](https://github.com/jeff1evesque/machine-learning/blob/6c438f85f8aa0d25e86a93a8c5fdab74e5ca5910/test/config/markdown.rb) to the following:

``` markdown
all
rule 'MD029', :style => :ordered
```

to explicitly ensure all rules were being implemented, before overriding `MD029`.  However, I still received the following [error](https://travis-ci.org/jeff1evesque/machine-learning/builds/90021747#L948-L958):

``` bash
...
$ mdl . --style test/config/markdown.md

/usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/style.rb:45:in `read': No such file or directory @ rb_sysopen - test/config/markdown.md (Errno::ENOENT)
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl/style.rb:45:in `load'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/lib/mdl.rb:16:in `run'
    from /usr/local/rvm/gems/ruby-2.2.3/gems/mdl-0.2.1/bin/mdl:10:in `<top (required)>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `load'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/mdl:23:in `<main>'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `eval'
    from /usr/local/rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `<main>'
The command "mdl . --style test/config/markdown.md" exited with 1.
...
```

**Comment by jeff1evesque** [comment]
Ah, I haven't been reading my errors very closely.  I may need to determine the relative working directory, with respect to the `mdl` command.  It seems the `test/config/` directory path that works for [other](https://github.com/jeff1evesque/machine-learning/blob/6635efd92424d00695b8ef7cc764a5c3d0aecf40/.travis.yml#L48-L49) linting processes in my [`.travis.yml`](https://github.com/jeff1evesque/machine-learning/blob/6635efd92424d00695b8ef7cc764a5c3d0aecf40/.travis.yml), doesn't work for the `mdl` command, respectively.

My current syntax seem correct, assuming `style_name` from [`configuration.md`](https://github.com/mivok/markdownlint/blob/master/docs/configuration.md#specifying-which-rules-mdl-processes), does not need to be quoted, or prefixed with a forward slash.

**Comment by jeff1evesque** [comment]
Seems my mistake was trying to reference `markdown.md`, when I should have been doing `markdown.rb`.
